client/src/components/Signup/SignupForm.jsx

```diff
@@ -0,0 +1,115 @@
+import { Formik, Form, Field } from 'formik';
+import React from 'react';
+import * as Yup from 'yup';
+
+const schema = Yup.object().shape({
+	firstname: Yup.string()
+		.required('Firstname is a required field')
+		.min(3, 'Firstname must be atleast 4 cahracters'),
+	lastname: Yup.string()
+		.required('Lastname is a required field')
+		.min(3, 'Lastname must be atleast 3 characters'),
+	username: Yup.string()
+		.required('Enter a username')
+		.min(5, 'username must be atleast 5 characters'),
+	dob: Yup.date(),
+	bio: Yup.string().min(5, 'Bio must be atleast 5 characters'),
+	location: Yup.string(),
+	email: Yup.string()
+		.required('Email is a required field')
+		.email('Invalid email format'),
+	password: Yup.string()
+		.required('Password is a required field')
+		.min(8, 'Password must be at least 8 characters'),
+});
+
+function SignupForm() {
+	return (
+		<Formik
+			validationSchema={schema}
+			initialValues={{ email: '', password: '' }}
```

Comment:
You need to add initial values for all fields. For the sign-up form, we don't need education, work, skills, and socials. 
We will have a feature to add them from the profile page as that would make the sign-up form too complicated.
I have removed them for now, but do add the initial values for the rest